src/librustc_error_codes/lib.rs

```diff
@@ -0,0 +1,26 @@
+//! This library is used to gather all error codes into one place. The goal
+//! being to make their maintenance easier.
+
+#[macro_export]
+macro_rules! register_diagnostics {
+    ($($ecode:ident: $message:expr,)*) => (
+        $crate::register_diagnostics!{$($ecode:$message,)* ;}
+    );
+
+    ($($ecode:ident: $message:expr,)* ; $($code:ident,)*) => (
+        pub static DIAGNOSTICS: &[(&str, &str)] = &[
+            $( (stringify!($ecode), $message), )*
+        ];
+
+        $(
```

Comment:
These constants shouldn't be necessary if we unlink things from the various crates as noted in a different comment of mine.

Reply:
Like I said, if we decide to do it, let's make it into another PR. I think this one is already big enough.